dataland-frontend/src/components/resources/companiesOnlySearch/CompaniesOnlySearchBar.vue

```diff
@@ -0,0 +1,111 @@
+<template>
+  <span class="p-fluid">
+    <span class="p-input-icon-left p-input-icon-right">
+      <i class="pi pi-search d-framework-searchbar-input-icon" aria-hidden="true" style="z-index: 20; color: #958d7c" />
+      <AutoComplete
+        inputId="company_search_bar_standard"
+        ref="autocomplete"
+        v-model="searchBarInput"
+        :suggestions="autocompleteArray"
+        :minLength="3"
+        optionLabel="companyInformation.companyName"
+        :autoOptionFocus="false"
+        placeholder="Search company by name or PermID"
+        inputClass="h-3rem d-framework-searchbar-input"
+        panelClass="d-framework-searchbar-panel"
+        style="z-index: 10"
+        @complete="searchCompanyName"
+        @item-select="pushToChooseFrameworkForDataUploadPageForItem"
+      >
+        <template #option="slotProps">
+          <i class="pi pi-search pl-3 pr-3" aria-hidden="true" />
+          <SearchResultHighlighter
+            :text="slotProps.option.companyInformation.companyName"
+            :searchString="latestValidSearchString"
+          />
+        </template>
+      </AutoComplete>
+    </span>
+  </span>
+</template>
+
+<script lang="ts">
+import AutoComplete from "primevue/autocomplete";
+import { StoredCompany } from "@clients/backend";
+import SearchResultHighlighter from "@/components/resources/frameworkDataSearch/SearchResultHighlighter.vue";
+import { defineComponent, inject, ref } from "vue";
+import Keycloak from "keycloak-js";
+import { useRoute } from "vue-router";
+import { assertDefined } from "@/utils/TypeScriptUtils";
+import { ApiClientProvider } from "@/services/ApiClients";
+
+export default defineComponent({
+  setup() {
+    return {
+      getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
+      autocomplete: ref(),
+    };
+  },
+  name: "ComapniesOnlySearchBar",
+  components: { AutoComplete, SearchResultHighlighter },
+
+  mounted() {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
```

Comment:
I assume there was no other way to prevent an eslint error?

Reply:
Could not fix that in a short amount of time.

Reply:
Ok, i resolve the conversion

Reply:
Let me try one more time today, maybe I was just too tired yesterday 😄

Reply:
I managed to fix it.   Let me fix this at other places in the code too, then I will push it and inform you here.

Reply:
I have fixed this case of eslint-disable, and I have also fixed it for the FrameworkDataSearchBar.vue

I also took a look at other places where we already use eslint-disables on usages of $refs, but I couldn't find a way to fix it.
The problem is that TypeScript always says "functionNameX() does not exist on type SomeVueComponent", even though the vue component SomeVueComponent has the method functionNameX().

Reply:
Ok, for me it is resolved